--- orig/net/netns/socks.go	2026-05-21 21:27:19.000000000 +0300
+++ tailscale_src/net/netns/socks.go	2026-06-06 04:04:42.984946555 +0300
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & contributors
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build !ios && !js && !android && !ts_omit_useproxy
+//go:build !ios && !js && !ts_omit_useproxy
 
 package netns
 
--- orig/net/netns/netns.go	2026-05-21 21:27:19.000000000 +0300
+++ tailscale_src/net/netns/netns.go	2026-06-06 04:04:42.984946555 +0300
@@ -119,6 +119,9 @@
 		panic("netns.FromDialer called with nil netMon")
 	}
 	if disabled.Load() {
+		if wrapDialer != nil {
+			return wrapDialer(d)
+		}
 		return d
 	}
 	d.Control = control(logf, netMon)
