--- orig/ipn/ipnlocal/local.go	2026-07-31 19:22:18.000000000 +0300
+++ tailscale_src/ipn/ipnlocal/local.go	2026-09-01 14:52:19.617816122 +0300
@@ -5724,8 +5724,9 @@
 
 	// Make a shallow copy of hostinfo so we can mutate
 	// at the Service field.
-	if f, ok := b.extHost.Hooks().ShouldUploadServices.GetOk(); !ok || !f() {
-		hi.Services = []tailcfg.Service{}
+	// Tailsocks: Allow advertising services on Android and include VIP services.
+	for _, vs := range b.vipServicesFromPrefsLocked(b.pm.CurrentPrefs()) {
+		hi.Services = append(hi.Services, tailcfg.Service{Proto: tailcfg.ServiceProto(vs.Name)})
 	}
 
 	// Don't mutate hi.Service's underlying array. Append to
--- orig/ipn/ipnlocal/serve.go	2026-07-31 19:22:18.000000000 +0300
+++ tailscale_src/ipn/ipnlocal/serve.go	2026-09-01 14:52:19.618457785 +0300
@@ -511,6 +511,7 @@
 			mak.Set(&services, svc, &tailcfg.VIPService{
 				Name:  svc,
 				Ports: config.ServicePortRange(),
+				Active: true,
 			})
 		}
 	}
