--- orig/ipn/ipnlocal/local.go	2026-05-21 21:27:19.000000000 +0300
+++ tailscale_src/ipn/ipnlocal/local.go	2026-05-24 18:53:16.991237649 +0300
@@ -5101,8 +5101,9 @@
 
 	// Make a shallow copy of hostinfo so we can mutate
 	// at the Service field.
-	if f, ok := b.extHost.Hooks().ShouldUploadServices.GetOk(); !ok || !f() {
-		hi.Services = []tailcfg.Service{}
+	// Tailsocks: Allow advertising services on Android and include VIP services.
+	for _, vs := range b.vipServicesFromPrefsLocked(b.pm.CurrentPrefs()) {
+		hi.Services = append(hi.Services, tailcfg.Service{Proto: tailcfg.ServiceProto(vs.Name)})
 	}
 
 	// Don't mutate hi.Service's underlying array. Append to
--- orig/ipn/ipnlocal/serve.go	2026-05-21 21:27:19.000000000 +0300
+++ tailscale_src/ipn/ipnlocal/serve.go	2026-05-24 18:53:16.995237649 +0300
@@ -510,6 +510,7 @@
 			mak.Set(&services, svc, &tailcfg.VIPService{
 				Name:  svc,
 				Ports: config.ServicePortRange(),
+				Active: true,
 			})
 		}
 	}
