--- orig/ipn/localapi/cert.go	2026-09-30 21:46:37.000000000 +0300
+++ tailscale_src/ipn/localapi/cert.go	2026-10-01 17:23:14.109065326 +0300
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & contributors
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build !ios && !android && !js && !ts_omit_acme
+//go:build !ios && !js && !ts_omit_acme
 
 package localapi
 
--- orig/ipn/localapi/disabled_stubs.go	2026-09-30 21:46:37.000000000 +0300
+++ tailscale_src/ipn/localapi/disabled_stubs.go	2026-10-01 17:23:14.109182865 +0300
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & contributors
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build ios || android || js
+//go:build ios || js
 
 package localapi
 
