--- orig/cmd/tailscaled/tailscaled_drive.go	2026-09-30 21:46:37.000000000 +0300
+++ tailscale_src/cmd/tailscaled/tailscaled_drive.go	2026-10-01 17:23:14.137092410 +0300
@@ -10,6 +10,7 @@
 	"fmt"
 
 	"tailscale.com/drive/driveimpl"
+	"tailscale.com/net/tsdial"
 	"tailscale.com/tsd"
 	"tailscale.com/types/logger"
 	"tailscale.com/wgengine"
@@ -19,7 +20,12 @@
 	subCommands["serve-taildrive"] = &serveDriveFunc
 
 	hookSetSysDrive.Set(func(sys *tsd.System, logf logger.Logf) {
-		sys.Set(driveimpl.NewFileSystemForRemote(logf))
+		sys.Set(driveimpl.NewFileSystemForRemote(logf, func() *tsdial.Dialer {
+			if d, ok := sys.Dialer.GetOK(); ok {
+				return d
+			}
+			return nil
+		}))
 	})
 	hookSetWgEnginConfigDrive.Set(func(conf *wgengine.Config, logf logger.Logf) {
 		conf.DriveForLocal = driveimpl.NewFileSystemForLocal(logf)
--- orig/cmd/tailscaled/tailscaled_windows.go	2026-09-30 21:46:37.000000000 +0300
+++ tailscale_src/cmd/tailscaled/tailscaled_windows.go	2026-10-01 17:25:05.208826935 +0300
@@ -48,6 +48,7 @@
 	"tailscale.com/logpolicy"
 	"tailscale.com/net/dns"
 	"tailscale.com/net/netmon"
+	"tailscale.com/net/tsdial"
 	"tailscale.com/net/tstun"
 	wintunconst "tailscale.com/tsconst/wintun"
 	"tailscale.com/tsd"
@@ -354,7 +355,12 @@
 	}
 	sys.Set(netMon)
 
-	sys.Set(driveimpl.NewFileSystemForRemote(log.Printf))
+	sys.Set(driveimpl.NewFileSystemForRemote(log.Printf, func() *tsdial.Dialer {
+		if d, ok := sys.Dialer.GetOK(); ok {
+			return d
+		}
+		return nil
+	}))
 
 	publicLogID, _ := logid.ParsePublicID(logID)
 	err = startIPNServer(ctx, log.Printf, publicLogID, sys)
--- orig/ipn/ipnlocal/local_test.go	2026-09-30 21:46:37.000000000 +0300
+++ tailscale_src/ipn/ipnlocal/local_test.go	2026-10-01 17:24:38.235313071 +0300
@@ -5471,7 +5471,7 @@
 				self.CapMap = tailcfg.NodeCapMap{nodecap.TaildriveShare: nil}
 				nm.SelfNode = self.View()
 				b.currentNode().SetNetMap(nm)
-				b.sys.Set(driveimpl.NewFileSystemForRemote(b.logf))
+				b.sys.Set(driveimpl.NewFileSystemForRemote(b.logf, nil))
 			}
 			b.mu.Unlock()
 
