--- orig/cmd/tailscaled/tailscaled_drive.go	2026-05-21 21:27:19.000000000 +0300
+++ tailscale_src/cmd/tailscaled/tailscaled_drive.go	2026-05-28 22:14:03.779809026 +0300
@@ -10,6 +10,7 @@
 	"fmt"
 
 	"tailscale.com/drive/driveimpl"
+	"tailscale.com/net/tsdial"
 	"tailscale.com/tsd"
 	"tailscale.com/types/logger"
 	"tailscale.com/wgengine"
@@ -19,7 +20,12 @@
 	subCommands["serve-taildrive"] = &serveDriveFunc
 
 	hookSetSysDrive.Set(func(sys *tsd.System, logf logger.Logf) {
-		sys.Set(driveimpl.NewFileSystemForRemote(logf))
+		sys.Set(driveimpl.NewFileSystemForRemote(logf, func() *tsdial.Dialer {
+			if d, ok := sys.Dialer.GetOK(); ok {
+				return d
+			}
+			return nil
+		}))
 	})
 	hookSetWgEnginConfigDrive.Set(func(conf *wgengine.Config, logf logger.Logf) {
 		conf.DriveForLocal = driveimpl.NewFileSystemForLocal(logf)
--- orig/cmd/tailscaled/tailscaled_windows.go	2026-05-21 21:27:19.000000000 +0300
+++ tailscale_src/cmd/tailscaled/tailscaled_windows.go	2026-05-28 22:14:15.251806476 +0300
@@ -49,6 +49,7 @@
 	"tailscale.com/net/dns"
 	"tailscale.com/net/netmon"
 	"tailscale.com/net/tstun"
+	"tailscale.com/net/tsdial"
 	"tailscale.com/tsd"
 	"tailscale.com/types/logger"
 	"tailscale.com/types/logid"
@@ -341,7 +342,12 @@
 	}
 	sys.Set(netMon)
 
-	sys.Set(driveimpl.NewFileSystemForRemote(log.Printf))
+	sys.Set(driveimpl.NewFileSystemForRemote(log.Printf, func() *tsdial.Dialer {
+		if d, ok := sys.Dialer.GetOK(); ok {
+			return d
+		}
+		return nil
+	}))
 
 	publicLogID, _ := logid.ParsePublicID(logID)
 	err = startIPNServer(ctx, log.Printf, publicLogID, sys)
--- orig/ipn/ipnlocal/local_test.go	2026-05-21 21:27:19.000000000 +0300
+++ tailscale_src/ipn/ipnlocal/local_test.go	2026-05-28 22:14:26.531803370 +0300
@@ -4593,7 +4593,7 @@
 				self.CapMap = tailcfg.NodeCapMap{tailcfg.NodeAttrsTaildriveShare: nil}
 				nm.SelfNode = self.View()
 				b.currentNode().SetNetMap(nm)
-				b.sys.Set(driveimpl.NewFileSystemForRemote(b.logf))
+				b.sys.Set(driveimpl.NewFileSystemForRemote(b.logf, nil))
 			}
 			b.mu.Unlock()
 
